articles/django-docker/index.md

```diff
@@ -0,0 +1,185 @@
+- `COPY ./requirements.txt /usr/src/app` copies the `requirements.txt` file into the work directory in the container.
+- `RUN pip install -r requirements.txt`
+ installs all the required modules for the `django_todo` application to run in the container.
+- `COPY . /usr/src/app` copies all the project source code to the working directory in the container.
+- `EXPOSE 8000` exposes port 8000 for access from other applications.
+- `CMD ["python", "manage.py", "runserver", "0.0.0.0:8000"]` sets the executable commands in the container.
+
+### Building Docker image
+To build the Docker image from the Dockerfile we created above, execute the command below.
+
+```bash
+$ docker build --tag django_todo:latest .
+```
+
+- `--tag` sets the tag for the image. For example, we are creating a Docker image from `python:3.8.3` which has the tag `alpine`. In our Docker image, `latest` is the tag set.
+- The trailing `.` indicates that the `Dockerfile` is within the current working directory.
+  
+To list all the available images on your computer, execute the command below.
+```bash
+$ docker image ls
+REPOSITORY    TAG            IMAGE ID       CREATED         SIZE
+django_todo   latest         6e06c89267f1   3 hours ago     147MB
+todo          latest         b10c177c6d58   4 hours ago     162MB
+<none>        <none>         2f418d359923   4 hours ago     162MB
+centos        latest         300e315adb2f   3 weeks ago     209MB
+python        3.8.3-alpine   8ecf5a48c789   6 months ago    78.9MB
+hello-world   latest         bf756fb1ae65   12 months ago   13.3kB
+
+```
+From the above list we see the `django_container` image that we have created.
+
+### Creating and running the Docker Container
+To build and run a Docker container from the Docker image we created above, run the command below.
+
+```bash
+$ docker run --name django_todo -d -p 8000:8000 django_todo:latest
+```
+- `--name` sets the name of the Docker container.
+- `-d` makes the image run in detached mode. The image is capable of running in the background.
+- `-p 8000:8000` maps port 8000 in the Docker container to port 8000 in the server.
+- `django_todo: latest` specifies which image is used to build the Docker container. 
+
+To list all the running Docker containers, execute the command below.
+```bash
+$ docker container ps
+CONTAINER ID   IMAGE                COMMAND                  CREATED       STATUS       PORTS                    NAMES
+d73306a9fb04   django_todo:latest   "python manage.py ru…"   3 hours ago   Up 3 hours   0.0.0.0:8000->8000/tcp   django_todo
+```
+On your browser visit `http://localhost:8000/api/v1/todo/` to confirm if the `django_todo` application is running in the container and you get a list of `todo` as a response.
+![Docker Hub image](/engineering-education/django-docker/docker-create.png)
+
+### Publishing Docker image to Docker Hub
+Docker Hub is a repository of container images that can be used to create other docker images.
+
```

Comment:
Docker images are used to create Docker containers. Please update your explanation.